python/ray/tune/suggest/variant_generator.py

```diff
@@ -55,6 +63,18 @@ def generate_trials(unresolved_spec, output_path=''):
                 args = parser.parse_args(to_argv(spec))
             except SystemExit:
                 raise TuneError("Error parsing args, see above message", spec)
+
+            new_config = copy.deepcopy(spec.get("config", {}))
+
+            # We hold the other resolved vars until suggestion is ready.
+            while search_alg is not None:
```

Comment:
Why does the generator function take in a search algorithm?

I would expect that it's the other way around. All trials come from the search algorithm

Reply:
This is just a helper function - we can move it into the ExistingVariant class.